sway-ir/src/value.rs

```diff
@@ -34,6 +34,7 @@ pub struct ValueContent {
 #[derive(Debug, Clone, DebugWithContext)]
 pub enum ValueDatum {
     Argument(BlockArgument),
+    Configurable(Constant),
```

Comment:
Did you consider just parameterising `Constant` to be configurable?  It feels like it'd be simpler than creating another `ValueDatum` variant, but perhaps there are blockers to this approach?

Reply:
My reasoning was that, if someone is writing an optimization pass that may touch constants, they have to know to check a flag called `is_configurable` if we were to parametrize `Constant`, and they have to do that everywhere. By making `Configurable` its own thing that is _initialized_ by a const but _not really_ a const, we force future developers to handle `Configurable` separately, or not at all, and existing passes continue to behave as they should (i.e. ignoring `Configurable`).